activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1620,13 +1620,17 @@ def build_where_clause(opts, rest = []) # :nodoc:
         case opts
         when String
           if rest.empty?
-            parts = [Arel.sql(opts)]
+            if Arel.arel_node?(opts)
+              parts = [opts]
+            else
+              parts = [Arel.sql(opts)]
+            end
```

Comment:
Shouldn't we just do that in `Arel.sql`? Right now it assumes it always receives a `String`, hence always wrap it, but we could make it return arel nodes unchanged.

Reply:
I tried searching for usages of `Arel.sql` that would pass a `SqlLiteral`, but it seems like there aren't many (any?) others than this one (I used `❯ rg '[^rf]\.sql\([^"%]'`).

I think it could make sense to do if the pattern was more common, but it seems like 90+% of the time a string is passed anyways.

Reply:
For a more definitive answer, I ran `ARCONN=sqlite3 bin/test` with this patch:

```
diff --git a/activerecord/lib/arel.rb b/activerecord/lib/arel.rb
index 738e80df35..cf82e60656 100644
--- a/activerecord/lib/arel.rb
+++ b/activerecord/lib/arel.rb
@@ -50,6 +50,7 @@ module Arel
   # Use this option only if the SQL is idempotent, as it could be executed
   # more than once.
   def self.sql(sql_string, *positional_binds, retryable: false, **named_binds)
+    raise if Nodes::SqlLiteral === sql_string
     if positional_binds.empty? && named_binds.empty?
       Arel::Nodes::SqlLiteral.new(sql_string, retryable: retryable)
     else
```

It fails on main but passes on this branch, so the two places touched by this PR are the only places we pass a `SqlLiteral` to `Arel.sql`.

Reply:
I guess my point was more for the future, and for general API design. e.g. `Array([]) # => []`.

Reply:
I feel a bit non-specifically wary of defining `Arel.sql` to pass-through all Nodes (while conceding that's functionally what we do in several call-sites already)... but regardless of that, I do agree it seems reasonable/sensible for it to pass-through its own results rather than re-wrapping them.

Reply:
Updated to pass through `SqlLiterals` 👍 I agree passing through other Nodes probably doesn't make sense (and I think `Arel.sql` doesn't even accept other nodes at the moment since they can't be coerced to String).